### PR TITLE
feat(rust, python): Supports series unique & arg_unique & n_unique for list

### PR DESCRIPTION
**Pull request by reswqa**

### CODE REVIEW

File: crates/polars-core/src/series/implementations/list.rs [hunk 1]
```diff
@@ -189,6 +189,55 @@ impl SeriesTrait for SeriesWrap<ListChunked> {
         self.0.has_validity()
     }
 
+    #[cfg(feature = "group_by_list")]
+    fn unique(&self) -> PolarsResult<Series> {
+        if !self.inner_dtype().is_numeric() {
+            polars_bail!(opq = unique, self.inner_dtype());
+        }
+        // this can called in aggregation, so this fast path can be worth a lot
```
Comment (mcrumiller) [inline crates/polars-core/src/series/implementations/list.rs:197]: typo, should say "this can **be** called"

Comment (mcrumiller) [reply crates/polars-core/src/series/implementations/list.rs:197]: (:D)

Comment (reswqa) [reply crates/polars-core/src/series/implementations/list.rs]: Yes, good catch!

**Comment by ritchie46** [comment]
I was always a bit opposed to this because I don't this we have the performance we want for this yet. But having the functionality is better than not.
 
 Thank you!

**Pull request merged**